pkg/eval/str/str.go

```diff
@@ -8,6 +8,313 @@ import (
 	"github.com/elves/elvish/pkg/eval"
 )
 
+//elvdoc:fn compare
+//
+// ```elvish
+// str:compare $a $b
+// ```
+//
+// Compare two strings and output an integer that will be 0 if a == b,
```

Comment:
Please use the verb form with "s", as if there is an implicit subject "this function". I have documented this in https://github.com/elves/elvish/blob/master/CONTRIBUTING.md#documenting-builtin-functions-and-variables.